doc/subsystems/networking/net-app-api.rst

```diff
@@ -0,0 +1,201 @@
+  user has set options like IP addresses, IEEE 802.15.4 channel etc. in the
+  project configuration file, then those settings are applied to the system
+  before application starts.
+
+* **Create** a simple TCP/UDP server or client application. The net-app API
+  has functions that make it easy to create a simple TCP or UDP based network
+  application. The net-app API also provides transparent TLS and DTLS support
+  for the application.
+
+The net-app API functionality is enabled by `CONFIG_NET_APP` option.
+The current net-app API implementation is still experimental and may change and
+improve in future releases.
+
+Initialization
+##############
+
+The net-app API provides `net_app_init()` function that can configure the
+networking subsystem for the application. Following configuration options
+control how the configuration is done:
+
+* **CONFIG_NET_APP_AUTO_INIT** automatically configures the system according
+  other configuration options. User does not need to call `net_app_init()` in
+  this case as that function will be automatically called when the system
+  boots. This option is enabled by default.
+
+* **CONFIG_NET_APP_INIT_TIMEOUT** tells how long to wait for the network
+  configuration during the system boot. For example if DHCPv4 is enabled, and
+  if the IPv4 address discovery takes long or the DHCPv4 server is not found,
+  the system will resume booting after this amount of seconds
+
+* **CONFIG_NET_APP_NEED_IPV6** tells that the application needs IPv6
+  connectivity. The `net_app_init()` will wait until it is able to setup
+  IPv6 address for the system before continuing. This means that the IPv6
+  duplicate address detection (DAD) has finished and the system has properly
+  setup the IPv6 address.
+
+* **CONFIG_NET_APP_NEED_IPV6_ROUTER** tells that the application needs IPv6
+  router connectivity. The `net_app_init()` will wait until it receives router
+  advertisement (RA) message from the IPv6 router before continuing.
+
+* **CONFIG_NET_APP_NEED_IPV4** tells that the application needs IPv4
+  connectivity. The `net_app_init()` will wait until it is able to setup
+  IPv4 address to the network subsystem.
+
+Setup
+#####
+
+Various system level network configuration options can be added to project
+configuration file. These settings are enabled by `CONFIG_NET_APP_SETTINGS`
+configuration option. This option is disabled by default. Note that for
+example the IPv6 options are only available if generic IPv6 support is enabled.
+
+* **CONFIG_NET_APP_MY_IPV6_ADDR** option sets a static IPv6 address to the
+  system. This is typically only useful in device testing as normally the
+  system should use SLAAC (IPv6 Stateless Address Auto Configuration) which
+  is enabled by default in the system. System can be configured to use
+  multiple IPv6 addresses, this is controlled by the
+  `CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT` configuration option.
+
+* **CONFIG_NET_APP_PEER_IPV6_ADDR** option tells what is the peer device
```

Comment:
There are uses of "tells" previously, but in this case I too would suggest, "option specifies the peer device IPv6 address."